utils/sampler/uniform_best.go

```diff
@@ -29,8 +29,12 @@ type uniformBest struct {
 func NewBestUniform(expectedSampleSize int) Uniform {
 	return &uniformBest{
 		samplers: []Uniform{
-			&uniformReplacer{},
-			&uniformResample{},
+			&uniformReplacer{
+				rng: globalRNG,
+			},
+			&uniformResample{
+				rng: globalRNG,
```

Comment:
nit: should we use the NewUniform constructor here?

Reply:
I think that would be pretty hard to understand which concrete type we are checking for (considering both of these structs are uniform samplers).